ebu_tt_live/carriage/filesystem.py

```diff
@@ -189,8 +190,27 @@ def __init__(self, dir_path, file_name_pattern):
         self._file_name_pattern = file_name_pattern
         self._counter = 0
 
-    def emit_document(self, document):
+    def _do_write_document(self, document):
         self._counter += 1
         filename = self._file_name_pattern.format(self._counter)
-        with open(os.path.join(self._dir_path, filename), 'w') as destfile:
+        filepath = os.path.join(self._dir_path, filename)
+        with open(filepath, 'w') as destfile:
             destfile.write(document.get_xml())
+            destfile.flush()
+        return filepath
+
+    def emit_document(self, document):
+        self._do_write_document(document)
+
+
+class RotatingFolderExport(SimpleFolderExport):
```

Comment:
Ok github truly is terrible the problem is that the explanation is there in the docstring just one line below: 
    """
    This carriage mechanism only keeps the last files that fit in its circular buffer. If a new file is written the
    oldest one is discarded. The size of the buffer can be specified. This is useful for the timeshift behaviour.
    """